modulo2/ciclo-vida-toda/src/App.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react'
+import styled from 'styled-components'
+import './styles.css'
+
+const TarefaList = styled.ul`
+  padding: 0;
+  width: 200px;
+`
+
+const Tarefa = styled.li`
+  text-align: left;
+  text-decoration: ${({completa}) => (completa ? 'line-through' : 'none')};
+`
+
+const InputsContainer = styled.div`
+  display: grid;
+  grid-auto-flow: column;
+  gap: 10px;
+`
+
+class App extends React.Component {
+    state = {
+      tarefas: [{
+        id: Date.now(), // Explicação abaixo
+        texto: 'Tarefa1',
+        completa: false // Indica se a tarefa está completa (true ou false)
+      },
+      {
+        id: Date.now(), // Explicação abaixo
+        texto: 'Tarefa2',
+        completa: true // Indica se a tarefa está completa (true ou false)
+      }],
+      inputValue: '',
+      filtro: "nenhum"
+    }
+
+  componentDidUpdate() {
+    localStorage.setItem("tasks", JSON.stringify(this.state.tarefas) )
+  };
+
+  componentDidMount() {
+    const taskString = localStorage.getItem("tasks")
+      const taksObj = JSON.parse(taskString) 
+      this.setState({tarefas: taksObj})
+  };
+  
```

Comment:
Nessa parte aqui Chris, quando você monta o componente, caso seu local Storage esteja vazio ele vai bugar o site por isso fica branco provavelmente, então pra "driblar" isso você pode por uma condicional ali dentro, ficando dessa forma:
~~~Js
  componentDidMount() {
    const taskString = localStorage.getItem("tasks")
    if (taskString) {
      const taksObj = JSON.parse(taskString) 
      this.setState({tarefas: taksObj})
      }
  };
~~~